presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java

```diff
@@ -206,7 +206,7 @@ public int getMaxSplitsPerSecond()
     @ConfigDescription("Throttles the maximum number of splits that can be assigned to tasks per second. 0 disables throttling.")
     public HiveConfig setMaxSplitsPerSecond(int maxSplitsPerSecond)
     {
-        this.maxSplitsPerSecond = Math.max(0, maxSplitsPerSecond);
+        this.maxSplitsPerSecond = Math.max(1, maxSplitsPerSecond);
```

Comment:
Remove the `Math.max()` here. The bean validation annotations like `@Min` work by first creating the object, which can be invalid, then validating all the properties at once, allowing to report every error in one shot. We don't want to use `max()` here as that will silently fix invalid configs, but the Presto philosophy is to fail, as that can hide an important mistake (the administrator may have intended to set this to a valid value).

Reply:
Great thanks, I didn't know about the validation using the annotation. Fixed :)